linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -237,7 +237,7 @@ MOCKABLE_FUNCTION(, int, threadpool_schedule_work, THANDLE(THREADPOOL), threadpo
 
 **SRS_THREADPOOL_LINUX_07_048: [** If reallocating the task array fails, `threadpool_schedule_work` shall fail and return a non-zero value. **]**
 
-**SRS_THREADPOOL_LINUX_07_049: [** `threadpool_schedule_work` shall initialize `pending_work_item_count_ptr` with `NULL` then copy the work function and work function context into insert position in the task array and assign `0` to the return variable to indicate success. **]**
+**SRS_THREADPOOL_LINUX_07_049: [** `threadpool_schedule_work` shall copy the work function and work function context into insert position in the task array and assign `0` to the return variable to indicate success. **]**
```

Comment:
>assign `0` to the return variable to indicate success [](http://example.com/codeflow?start=160&length=53)

and return 0.

Reply:
I see this recurring and it seems that it is not going away, so shall wait for it. We can discuss in person what the problem is, but I think it is better to deal with it once and for all.

Reply:
Hi Dan,

Thanks for the comments. The changes you are mentioning was done in the old commits on request of other reviewers. 
https://github.com/Azure/c-pal/pull/373/files/04ed1648d62fe6742bd7056d17c3a43a8175e18a#r1775937979
Please check above link to see the comments.

Since this story is really to unblock anyone using the old `threadpool_schedule_work` or someone facing failing tests, I sent the code review with minimal changes.

Can we address your other comments in the `threadpool_create_work_item` work that we will have to do later?

Reply:
@nishudeshu The short answer is no, we should address these comments now. Kicking them down the road does not buy us anything and there is a high chance they would remain like this in the code, which would be less than ideal.

Strictly on the 2 points:
1.  assign `0` to the return variable to indicate success => I have explained why we think 'shall return 0' is better. Changing this is trivial and we should not spend lots of energy in debating this. I assume you understood the reasoning already in the previous code review, but if that's not the case, I can explain it again.

2. The comments about pending_work_item_count_ptr in the code make it clear that even if you fix the current helgrind failure, other issues exist and they will surface again. Let's address them now, when the context is in front of us!